lib/ansible/modules/web_infrastructure/letsencrypt.py

```diff
@@ -419,14 +419,16 @@ def __init__(self, module):
 
         # Create a key file from content, key (path) and key content are mutually exclusive
         if self.key_content is not None:
-            _, tmpsrc = tempfile.mkstemp()
-            module.add_cleanup_file(tmpsrc)  # Ansible will delete the file on exit
```

Comment:
ehm, this should stay here. ;)

Reply:
Interestingly, I don't think it's done in the code above. Instead, it rather franticly tries to do a lot of checks and remove the file. It actually also tries to remove the file if it doesn't exist (https://github.com/ansible/ansible/pull/35278/files#diff-914391d9e58eafc28fbb8234b9e00e17R335) ... I think that'll crash the module.

Reply:
Oooops, yes, that should've stayed!

Reply:
@thomwiggers: indeed:
``` .py
>>> import os
>>> os.remove('adsfsafdsa')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: 'adsfsafdsa'
```
I'll add a second commit fixing that.

Reply:
I decided to add a `try/except` ignoring the exception, as this case should never happen anyway in any sane circumstances (after all, we just wrote something to that file).